core/src/main/java/org/jdbi/v3/core/array/SqlArrayArgument.java

```diff
@@ -18,34 +18,27 @@
 import java.sql.SQLException;
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.Collection;
 import java.util.List;
-
-import org.jdbi.v3.core.statement.StatementContext;
 import org.jdbi.v3.core.argument.Argument;
 import org.jdbi.v3.core.internal.IterableLike;
+import org.jdbi.v3.core.statement.StatementContext;
 
 class SqlArrayArgument<T> implements Argument {
-
     private final String typeName;
     private final Object[] array;
 
-    @SuppressWarnings("unchecked")
     SqlArrayArgument(SqlArrayType<T> arrayType, Object newArray) {
         this.typeName = arrayType.getTypeName();
 
-        List<Object> elements = new ArrayList<>(
-                newArray.getClass().isArray() ? Array.getLength(newArray) : 10);
-        IterableLike.of(newArray).forEachRemaining(
-                e -> elements.add(arrayType.convertArrayElement((T) e)));
+        List<Object> elements = new ArrayList<>(newArray.getClass().isArray() ? Array.getLength(newArray) : 10);
+        IterableLike.of(newArray).forEachRemaining(e -> {
```

Comment:
What's going on in this refactoring?  It looks the same but it's way wordier now.  This doesn't seem like an improvement to me unless I'm missing something.

Reply:
If you open the old code, not being familiar with it, it takes some serious reading to figure out. The lookup variable just obscured the process. The whole method top to bottom just goes "this is either an array or a list, we're gonna cast it to whichever it is, and call 2 different flows to finish up depending on which it is". For all intents and purposes the method was just 2 functionally distinct methods clobbered together to save on lines of text, and not really structured to express its binary nature either.

I just factored the 2 methods apart and sort of normalized them. The fact that they look very different now is a clear sign they shouldn't have been writting as one whole. They happen to share a few variables and the null part, but that's all they share.

It's all about clarifying intent, making expression match function.

Reply:
oh, and IDE support becomes pretty crap too when a variable is inferred to be "either this or that, who knows" :p

Reply:
Oh, this part. Reducing the scope of the SuppressWarnings

Reply:
It turned two lines into five lines, that sure seems wordier to me.  You even extracted out a new variable `cast` that is totally useless.  This PR would have been a slam dunk if it just had the relevant fix at hand...

Reply:
?? I only _added_ 1 line of minimal code, by extracting the cast to keep the scope of the SuppressWarnings to a minimum. The only other addition is a pair of brackets because multiline lambda. I really don't see how it's wordier. It's not more code. I literally just moved the suppresswarnings and extracted a variable for good practice.